### PR TITLE
amam/refactor_all_components

### PR DESCRIPTION
**Pull request by mustofa-binary**
# Description

refactoring all components includes removing unused components, add new features to some components

Changes:
- Create `Box` components that includes css-parameters built in functions like margins paddings, etc.
- Combine `Text` and `Header` base css like font-size , lh, to `BaseElement` css
- Fix misconduct code on styled component on all components like usage of nested div inside styled component which can be extracted to a new styled component
- Removes some of the unused components Modal, LinearLoader, Hero, etc.
- refactor margin and padding components from `margin={{ top: '12rem', horizontal: '8rem' }}` to `m='0 8rem' mt='12rem'`
- Remove unused modal component usages
- Apply changes Text and Header to all pages and components

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [x] Refactor code
-   [ ] Update translation text
-   [ ] Breaking change
-   [ ] Dependency update
-   [ ] This change requires a documentation update
-   [ ] Release

### CODE REVIEW

**Comment by arash-fs** [comment]
do think it's not better to keep the modal component? maybe we should use it again.

File: src/pages/help-centre/dmt5.js [hunk 1]
```diff
@@ -42,9 +43,9 @@ const WhatIsDMT5 = () => (
 )
 const DifferenceDMT5DTrader = () => (
     <ArticleWrapper>
-        <Header as="h4" margin=" 0 0 2.4rem 0" lh="1.5">
+        <StyledHeader as="h4">
```
Comment (arash-fs) [inline src/pages/help-centre/dmt5.js]: here we have `lh='1.5` in Header after using StyledHeader we did not add `lh` to the component, please make sure that line-height is added to the StyledHeader

**Comment by arash-fs** [comment]
Please check the StyledHeader in codes. I find the Header had line-height style but when you use StyledHeader instead of that didn't use line-height. please be sure about the line-height applies to the StyledHeader component. I mentioned in codes but there are too many of them.

File: src/pages/help-centre/security.js [hunk 2]
```diff
@@ -15,11 +15,13 @@ const ArticleWrapper = styled.div`
...
+const StyledHeader = styled(Header)`
+    margin-bottom: 2.4rem;
+`
+
 const VerifyAccount = () => (
     <ArticleWrapper>
-        <Header as="h4" margin=" 0 0 2.4rem 0" lh="1.5">
-            {localize('Do I need to verify my Deriv account?')}
-        </Header>
+        <StyledHeader as="h4">{localize('Do I need to verify my Deriv account?')}</StyledHeader>
```
Comment (arash-fs) [inline src/pages/help-centre/security.js]: Also here, please check the line-height

File: src/pages/partners/_deriv-ib-programme.js [hunk 3]
```diff
@@ -36,9 +32,8 @@ const BackButton = styled(Button)`
     color: var(--color-black);
 `
 const StyledText = styled(Text)`
-    font-size: ${props => (props.font_size ? props.font_size : '1.4rem')};
-    line-height: ${props => (props.lh ? props.lh : '1.15')};
-    margin-bottom: ${props => props.margin_bottom || ''};
+    font-size: 1.4rem;
+    line-height: 1.15;
```
Comment (arash-fs) [inline src/pages/partners/_deriv-ib-programme.js]: we have replicated the line-height in this component, in line 469 we declare it inline too.

File: src/pages/help-centre/dmt5.js (hunk 1 above)
Comment (mustofa-binary) [reply src/pages/help-centre/dmt5.js]: i put default lh for header is 1.5

File: src/pages/help-centre/security.js (hunk 2 above)
Comment (mustofa-binary) [reply src/pages/help-centre/security.js]: ive put default header have line-height of 1.5

File: src/components/containers/container.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import styled from 'styled-components'
 import device from 'themes/device'
+import { Margins, Paddings } from 'themes/function'
 
 const Container = styled.div`
```
Comment (oskar-binary) [inline src/components/containers/container.js]: Why not styled(Box)?

File: src/components/containers/flex.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import styled from 'styled-components'
+import { Margins, Paddings } from 'themes/function'
 
 const Flex = styled.div`
```
Comment (oskar-binary) [inline src/components/containers/flex.js]: and here too

**Comment by oskar-binary** [comment]
Could you update this with the latest changes from dev and remove `package-lock.json` (+node_modules) and do a fresh `npm i`? So we can see if the build time is affected.

**Comment by ashkanx** [comment]
@mustofa-binary conflicts

**Comment by oskar-binary** [comment]
LGTM 👍

File: src/components/custom/_signup-public.js [hunk 6]
```diff
@@ -290,7 +291,7 @@ const SignupPublic = ({
             </SignupFormWrapper>
             <BackgroundWrapper direction="column" ai="center">
                 <LinkFlex ai="center" onClick={redirectToDerivApp}>
-                    <StyledHeader font_size="2.8rem" width="28.2rem" align="center" color="grey-8">
+                    <StyledHeader size="2.8rem" max_width="28.2rem" align="center" color="grey-8">
```
Comment (oskar-binary) [inline src/components/custom/_signup-public.js]: Should be `mw` here?
```suggestion
                    <StyledHeader size="2.8rem" mw="28.2rem" align="center" color="grey-8">
```

Comment (oskar-binary) [reply src/components/custom/_signup-public.js]: but in other places its `max_width`, should we stick with `mw`?

Comment (mustofa-binary) [reply src/components/custom/_signup-public.js]: it should be mw, let me check

Comment (mustofa-binary) [reply src/components/custom/_signup-public.js]: wait, if max_width is mw, then what is min_width?

Comment (oskar-binary) [reply src/components/custom/_signup-public.js]: 🔢 true, better stick with max_width and min_width

**Pull request merged**